lib/lltz_michelson/lltz_michelson.ml

```diff
@@ -87,4 +88,97 @@ and convert_constant (const: LLTZ.E.constant) : Michelson.Ast.t =
     | Bls12_381_g1 _ -> Michelson.T.bls12_381_g1
     | Bls12_381_g2 _ -> Michelson.T.bls12_381_g2
     | Bls12_381_fr _ -> Michelson.T.bls12_381_fr
-    | Signature _ -> Michelson.T.signature
\ No newline at end of file
+    | Signature _ -> Michelson.T.signature
+
+let convert_primitive (prim: LLTZ.P.t) : Michelson.Ast.t =
+  let open Michelson.Ast.Instruction in
+    match prim with
+    | Amount -> amount
+    | Balance -> balance
+    | Chain_id -> chain_id
+    | Level -> level
+    | Now -> now
+    | Self opt -> (match opt with Some str -> self | None -> self)
```

Comment:
| Self Some str -> self
| Self None -> self

But don't you have a warning for the unused variable `str`?

Reply:
```suggestion
    | Self opt -> failwith
```

Reply:
right, changed it to failwith for now as its another case with these string options which seem wrongly defined in the ir.